pkg/resourcegroup/controllers/resourcegroup/resourcegroup_controller.go

```diff
@@ -205,20 +193,29 @@ func (r *reconciler) endReconcilingStatus(
 	}()
```

Comment:
Is this goroutine being leaked in the case of the `reconcileTimeout` being exceeded? It doesn't look like we wait for this goroutine to exit in that case.

Reply:
Yes. It's not being cancelled because it has no context... Unrelated bug.

Reply:
IMO this needs to be rewritten so that it calculates the resource statuses asynchronously, caches them, and just updates the RG status based on the cache, but that's a bigger effort I haven't gotten around to.

Reply:
The problem is that if this controller is triggered by resource updates, then the cache would almost always be somewhat stale, depending on whether a managed object update triggered it or a RG update triggered it. So it needs to be sequenced AFTER updating the cache to reflect the status of whatever triggered it. That way it wouldn't need to recompute all the resource statused on every reconcile pass.